lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -1102,12 +1102,20 @@ MDNode *SPIRVToLLVMDbgTran::transTypeTemplate(const SPIRVExtInst *DebugInst) {
 DINode *SPIRVToLLVMDbgTran::transImportedEntry(const SPIRVExtInst *DebugInst) {
   using namespace SPIRVDebug::Operand::ImportedEntity;
   const SPIRVWordVec &Ops = DebugInst->getArguments();
-  assert(Ops.size() >= OperandCount && "Invalid number of operands");
-  DIScope *Scope = getScope(BM->getEntry(Ops[ParentIdx]));
-  SPIRVWord Line =
-      getConstantValueOrLiteral(Ops, LineIdx, DebugInst->getExtSetKind());
-  DIFile *File = getFile(Ops[SourceIdx]);
-  auto *Entity = transDebugInst<DINode>(BM->get<SPIRVExtInst>(Ops[EntityIdx]));
+  // FIXME: 'OpenCL/bugged' version is kept because it's hard to remove it
+  // It's W/A for missing 2nd index in OpenCL's implementation
+  const SPIRVWord OffsetIdx = isNonSemanticDebugInfo(DebugInst->getExtSetKind())
+                                  ? OperandCount - NonSemantic::OperandCount
+                                  : 0;
+
+  assert(Ops.size() >= (OperandCount - OffsetIdx) &&
```

Comment:
There are no optional operands for this instruction - we can check for `==` instead

Reply:
Applied, thanks!